apps/files_encryption/lib/helper.php

```
@@ -281,55 +281,68 @@ public static function getUser($path) {
 
-		// it is not a file relative to data/user/files
-		if (count($split) < 2 || ($split[1] !== 'files' && $split[1] !== 'cache')) {
-			return false;
-		}
+		if (count($split) > 3 && (
+			$split[2] === 'files' || $split[2] === 'files_versions' || $split[2] === 'cache')) {
 
-		$user = $split[0];
+			$user = $split[1];
 
-		if (\OCP\User::userExists($user)) {
-			return $user;
+			if (\OCP\User::userExists($user)) {
+				return $user;
+			}
 		}
 
-		return false;
+		throw new Exception\EncryptionException('Could not determine user', Exception\EncryptionException::GENERIC);
 	}
 
 	/**
 	 * get path to the corresponding file in data/user/files if path points
-	 *        to a version or to a file in cache
-	 * @param string $path path to a version or a file in the trash
+	 * to a file in cache
+	 *
+	 * @param string $path path to a file in cache
 	 * @return string path to corresponding file relative to data/user/files
 	 */
-	public static function getPathToRealFile($path) {
-		$trimmed = ltrim($path, '/');
-		$split = explode('/', $trimmed);
-		$result = false;
-
-		if (count($split) >= 3 && ($split[1] === "files_versions" || $split[1] === 'cache')) {
-			$sliced = array_slice($split, 2);
-			$result = implode('/', $sliced);
-			if ($split[1] === "files_versions") {
-				// we skip user/files
-				$sliced = array_slice($split, 2);
-				$relPath = implode('/', $sliced);
-				//remove the last .v
-				$result = substr($relPath, 0, strrpos($relPath, '.v'));
-			}
-			if ($split[1] === "cache") {
-				// we skip /user/cache/transactionId
-				$sliced = array_slice($split, 3);
-				$result = implode('/', $sliced);
-				//prepare the folders
-				self::mkdirr($path, new \OC\Files\View('/'));
-			}
-		}
+	public static function getPathFromCachedFile($path) {
+		$normalized = \OC\Files\Filesystem::normalizePath($path);
+		$split = explode('/', $normalized);
+
+		// we skip /user/cache/transactionId
+		$sliced = array_slice($split, 4);
```

Add a count for `$split`? - While currently it might always be called in a way with proper paths this might change in future, maybe due to a bug or whatever.

I'd throw an error in case the array does not contain enough elements. Easiest approach to catch potential bugs in the future.